### PR TITLE
Developing Super-Fast SPAs with Choo

### PR DESCRIPTION
**Pull request by HoverBaum**
A blogpost about building a simple and fun application in Choo and using Auth0 to authenticate Users.

Not quite sure on how to use images in blogposts here. How would I do that?

### CODE REVIEW

File: articles/2018-09-27-securing-choo.markdown [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+description: "Follow along as we secure a simple, Choo based application using Auth0."
+date: "2018-09-27 08:30"
+author:
+  name: "Hendrik Wallbaum"
+  url: "hoverbaum"
+  mail: "mail@hendrikwallbaum.de"
+  avatar: "https://twitter.com/hoverbaum/profile_image?size=original"
+---
+
+**TL;DR:** Today we will get familiar with Choo as a super slim and fun frontend framework, build a friendly application and secure it using Auth0. Check the [working demo](https://auth0-secured-choo.netlify.com/) and [GitHub repo](https://github.com/HoverBaum/choo-auth0).
```
Comment (brunokrebs) [inline articles/2018-09-27-securing-choo.markdown]: Apparently your app is not working.

File: articles/2018-09-27-securing-choo.markdown [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+
+# A simple and fun Choo app secured with Auth0
+
+Securing an application and authenticating users is a crucial part for every modern web app. Luckily the times of having to build your own authentication solution using PHP and MySQL databases are over. In this time an age we have great services that specialize in providing authentication and authorization as a service. By focusing on this single task services are able to provide a far better solution than anybody implementing a user system as part of their app ever could. And it is super easy to integrate these services thus everyone benefits.
```
Comment (brunokrebs) [inline articles/2018-09-27-securing-choo.markdown]: this time an age?

File: articles/2018-09-27-securing-choo.markdown [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+
+To demonstrate how easily you can integrate such a service we are going to build a simple Choo (like the train goes choo choo) application. Users will be able to sign up and log in. After Logging in they will see an identiticon (liek teh standard GitHub avatars) generated just for them. Instead of building the user system ourselfes we will let [Auth0](https://auth0.com/) handle that for us.
```
Comment (brunokrebs) [inline articles/2018-09-27-securing-choo.markdown]: > liek teh standard GitHub avatars

I see some typos here

File: articles/2018-09-27-securing-choo.markdown [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+
+Next up is the logic to handle our application being called after a successful login. For that we are going to add a piece of code at the bottom of our store. Again inspired by the quick start guide Auth0 provides. Just put the code at the buttom of your `authState` function.
+
+```javascript
+// If we are serverside at this point return ebcause now we will look at authentication on the client.
+if (typeof window === 'undefined') return
+
+// Check local storage if we are logged in.
+const storedIdToken = window.localStorage.idToken
+const storedUserId = window.localStorage.userId
```
Comment (brunokrebs) [inline articles/2018-09-27-securing-choo.markdown]: we are actually not encouraging the usage of localStorage anymore... this has become a security issue. Nowadays, we keep tokens and sensitive data like those in memory. Please, check the _Securing your React App with Auth0_ section of this article: https://auth0.com/blog/react-tutorial-building-and-securing-your-first-app/

As you will probably realize, if your users refresh the page, their tokens will get lost. To keep users signed in, please, check this section on the same article: Keeping Users Signed In after a Refresh

File: articles/2018-09-27-securing-choo.markdown (hunk 1 above)
Comment (HoverBaum) [reply articles/2018-09-27-securing-choo.markdown]: Awesome catch, can't believe I missed that. Working now :)

**Pull request closed without merging**